cartridge/shop/views.py

```diff
@@ -282,6 +299,15 @@ def checkout_steps(request, form_class=OrderForm, extra_context=None):
                 # and send the order receipt email.
                 order = form.save(commit=False)
                 order.setup(request)
+
+                if (not settings.SHOP_PAYMENT_STEP_ENABLED or
+                        order.total == 0):
```

Comment:
I'm confused. Doesn't the `step+=1` part skip the payment step entirely? When would `order.total == 0` at this point?